clients/imodels-client-management/src/base/internal/DefaultHttpRequestRetryPolicy.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { GetSleepDurationInMsParams, HttpRequestRetryPolicy, OnFailedRequestParams, ShouldRetryParams, isIModelsApiError } from "../types";
+
+import { BackoffAlgorithm } from "./ExponentialBackoffAlgorithm";
+import { isTransientHttpStatusCode } from "./UtilityFunctions";
+
+/** Default implementation for {@link HttpRequestRetryPolicy}. */
+export class DefaultHttpRequestRetryPolicy implements HttpRequestRetryPolicy {
+  private readonly _retries: number;
+  private readonly _backoffProvider: BackoffAlgorithm;
+
+  public constructor(
+    backoffProvider: BackoffAlgorithm,
+    retries: number = 2
+  ) {
+    this._retries = retries;
+    this._backoffProvider = backoffProvider;
+  }
+
+  public onFailedRequest(_: OnFailedRequestParams): void {}
```

Comment:
we ourselves do not use this function so I would not expect anyone else to use this. If it's not needed now, remove this, we can add it later. For logging purposes `shouldRetry` could be reused

Reply:
updated